wren-main/src/main/java/io/wren/main/wireprotocol/message/SimpleQuery.java

```diff
@@ -0,0 +1,110 @@
+import static io.wren.main.wireprotocol.Utils.readCString;
+import static io.wren.main.wireprotocol.message.MessageUtils.isIgnoredCommand;
+import static io.wren.main.wireprotocol.message.MessageUtils.sendHardWiredSessionProperty;
+
+public class SimpleQuery
+        implements Commit
+{
+    public static SimpleQuery simpleQuery(ByteBuf buffer)
+    {
+        String statement = readCString(buffer);
+        checkArgument(statement != null, "query must not be null");
+        return new SimpleQuery(statement);
+    }
+
+    private static final Logger LOG = Logger.get(SimpleQuery.class);
+
+    private final String statement;
+
+    private SimpleQuery(String statement)
+    {
+        this.statement = statement;
+    }
+
+    @Override
+    public void commit(CompletableFuture<?> planned, Channel channel, WireProtocolSession session)
+    {
+        planned.thenCompose(v -> handleSimpleQuery(channel, session));
+    }
+
+    private CompletableFuture<?> handleSimpleQuery(Channel channel, WireProtocolSession session)
+    {
+        List<String> queries = QueryStringSplitter.splitQuery(statement);
+
+        CompletableFuture<?> composedFuture = CompletableFuture.completedFuture(null);
+        for (String query : queries) {
+            composedFuture = composedFuture.thenCompose(result -> handleSingleQuery(query, channel, session));
+        }
+        return composedFuture.whenComplete(new ReadyForQueryCallback(channel, TransactionState.IDLE));
+    }
+
+    private CompletableFuture<?> handleSingleQuery(String statement, Channel channel, WireProtocolSession wireProtocolSession)
+    {
+        if (statement.isEmpty() || ";".equals(statement.trim())) {
+            ResponseMessages.sendEmptyQueryResponse(channel);
+            return CompletableFuture.completedFuture(null);
+        }
+        if (isIgnoredCommand(statement)) {
+            sendHardWiredSessionProperty(channel, statement);
+            ResponseMessages.sendCommandComplete(channel, statement, 0);
+            return CompletableFuture.completedFuture(null);
+        }
+        try {
+            wireProtocolSession.parse("", statement, ImmutableList.of());
+            wireProtocolSession.bind("", "", ImmutableList.of(), null);
+            Optional<ConnectorRecordIterator> iterator = wireProtocolSession.execute("").join();
+            if (iterator.isEmpty()) {
+                sendHardWiredSessionProperty(channel, statement);
+                ResponseMessages.sendCommandComplete(channel, statement, 0);
+                return CompletableFuture.completedFuture(null);
+            }
+            ResultSetSender resultSetSender = new ResultSetSender(
+                    statement,
+                    channel,
+                    iterator.get(),
+                    0,
+                    0,
+                    null);
+            ResponseMessages.sendRowDescription(channel, wireProtocolSession.describePortal("").get(), null);
```

Comment:
It should use `resultSetSender.getCloumns()` instead of `wireProtocolSession.describePortal("")`, Am I right?

Reply:
Yes, you're right. It can use getColumn instead.